libs/goaccumulate/goaccumulate.go

```diff
@@ -0,0 +1,247 @@
+package goaccumulate
+
+import (
+	"bytes"
+	"context"
+	"crypto/ed25519"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	"github.com/AccumulateNetwork/accumulate/client"
+	"github.com/AccumulateNetwork/accumulate/cmd/cli/db"
+	api2 "github.com/AccumulateNetwork/accumulate/internal/api/v2"
+	url2 "github.com/AccumulateNetwork/accumulate/internal/url"
+	"github.com/AccumulateNetwork/accumulate/protocol"
+	"github.com/mdp/qrterminal"
+)
+
+var (
+	Client         = client.NewAPIClient()
+	Db             db.DB
+	WantJsonOutput = false
+	TxPretend      = false
+)
+
+var (
+	BucketAnon     = []byte("anon")
+	BucketAdi      = []byte("adi")
+	BucketKeys     = []byte("keys")
+	BucketLabel    = []byte("label")
+	BucketMnemonic = []byte("mnemonic")
+)
+
+// -------------------------------------------------------------------------------
+// Tokens
+// -------------------------------------------------------------------------------
+
+func GetToken(url string) {
+	var res api2.QueryResponse
+
+	params := api2.UrlQuery{}
+	params.Url = url
+
+	data, err := json.Marshal(&params)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := Client.RequestV2(context.Background(), "query", json.RawMessage(data), &res); err != nil {
+		log.Fatal(err)
+	}
+
+	str, err := json.Marshal(res)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println(string(str))
+
+}
+
+func CreateToken(url string, symbol string, precision string, signer string) {
+	fmt.Println("Creating new token " + symbol)
+}
+
+// -------------------------------------------------------------------------------
+// Accounts
+// -------------------------------------------------------------------------------
+
+func GetAccount(url string) (string, error) {
+	var rez api2.QueryResponse
+	u, err := url2.Parse(url)
+	if err != nil {
+		return "", fmt.Errorf("%q is not a valid Accumulate URL: %v\n", url, err)
+	}
+
+	params := api2.UrlQuery{}
+	params.Url = u.String()
+
+	data, err := json.Marshal(&params)
+	if err != nil {
+		return "", err
+	}
+
+	if err := Client.Request(context.Background(), "query", json.RawMessage(data), &rez); err != nil {
+		return PrintJsonRpcError(err)
+	}
+
+	return PrintQueryResponseV2(&rez)
+}
+
+func QrAccount(s string) (string, error) {
+	u, err := url2.Parse(s)
+	if err != nil {
+		return "", fmt.Errorf("%q is not a valid Accumulate URL: %v\n", s, err)
+	}
+
+	b := bytes.NewBufferString("")
+	qrterminal.GenerateWithConfig(u.String(), qrterminal.Config{
+		Level:          qrterminal.M,
+		Writer:         b,
+		HalfBlocks:     true,
+		BlackChar:      qrterminal.BLACK_BLACK,
+		BlackWhiteChar: qrterminal.BLACK_WHITE,
+		WhiteChar:      qrterminal.WHITE_WHITE,
+		WhiteBlackChar: qrterminal.WHITE_BLACK,
+		QuietZone:      2,
+	})
+
+	r, err := ioutil.ReadAll(b)
+	return string(r), err
+}
+
+func CreateAccount(url string, args []string) (string, error) {
```

Comment:
the arguments passed to this function should not be an array of strings.  should be either the data structure of explicit parameters

Reply:
perhaps explore the use of a builder pattern instead.